src/main/httpCommandServer.ts

```diff
@@ -118,32 +123,80 @@ export class HttpCommandServer {
     return this.dispatchTable[version]?.[method]?.[commandName];
   }
 
-  listSettings(request: http.IncomingMessage, response: http.ServerResponse) {
-    const settings = this.commandWorker?.getSettings();
+  async listSettings(request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await new Promise((resolve) => {
+      const settings = this.commandWorker?.getSettings();
 
-    if (settings) {
-      response.writeHead(200, { 'Content-Type': 'text/plain' });
-      response.write(settings);
+      if (settings) {
+        console.log('listSettings: succeeded 200');
+        response.writeHead(200, { 'Content-Type': 'text/plain' });
+        response.write(settings);
+      } else {
+        console.log('listSettings: failed 200');
+        response.writeHead(404, { 'Content-Type': 'text/plain' });
+        response.write('No settings found');
+      }
+      resolve();
+    });
+  }
+
+  /**
+   * Allow the parameters to come in both via URL parameters (non-traditional) and in a request body.
+   * @param request
+   * @param response
+   */
+  async updateSettings(request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    const url = new URL(request.url as string, `http://${ request.headers.host }`);
+    // TODO: Likely we'll be dropping support for searchParams here.
+    const searchParams = url.searchParams;
+    const chunks: Buffer[] = [];
+
+    // Read in the request body
+    for await (const chunk of request) {
+      chunks.push(chunk);
+    }
+    const data = Buffer.concat(chunks).toString();
+    // Deliberately throw a (500) exception if the data exists and isn't valid JSON
```

Comment:
Wouldn't that be a 400 (bad request)?

Reply:
Yes, because the server should allow for ill-formed JSON. The 500 codes are reserved for unanticipated exceptions, in other words, bugs.

Had to rewrite this part of the code almost completely because of the need for a try/catch block. And I realized I wasn't checking for empty or oversized request blocks. We didn't specify just what oversized is, but for this method anything more than 2K seems like either client error or an attempted DoS.